### PR TITLE
Implementation of eRMSD (AA), eRMSD (CG) and  RMSD (Kabsch) [all unweighted]

### PR DESCRIPTION
**Pull request by yihengwuKP**
Hi,  I've implemented [eRMSD](https://doi.org/10.1093/nar/gku972) in AA or CG mappings. I've also incorporated Yezhi's implementation of Kabsch RMSD into the code @yzjin . I've implemented tests for all of them. 

The eRMSD requires knowing the C2/C4/C6 of the six-membered ring in the base. The implementation of eRMSDCG is for the situation when we only know the other three sites in the base, and we need to first infer the position of C2/C4/C6 before we calculate the eRMSD. 

The testing of eRMSD (AA) was done by comparing with the calculation exactly done by [barnaba](https://github.com/srnas/barnaba). The testing of eRMSDCG was done by comparing with the eRMSD so it's not exact but very close. The testing for RMSD is done by precalculated examples as in [test kabsch](https://github.com/charnley/rmsd/blob/master/tests/test_kabsch.py) as mentioned in https://github.com/SSAGESLabs/PySAGES/issues/305

Please let me know if there are any suggestions/comments!

### CODE REVIEW

File: pysages/colvars/orientation.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+        Weights of the selected atom positions
+    """
+    P = fitted_positions(r1, w_0)
+    U = kabsch(P, Q)
+    optimal_P = np.dot(P, U)
+    error = np.sqrt(np.sum(np.square(optimal_P - Q)) / P.shape[0])
+    return error
+
+
+class RMSD_Kabsch(CollectiveVariable):
```
Comment (pabloferz) [inline pysages/colvars/orientation.py:96]: I would prefer to have just one RMSD class and have a class argument for indicating the algorithm to estimate the optimal rotation matrix. Something along these lines:
```python
class RMSD(CollectiveVariable):
    def __init__(self, indices, references, weights=None, optimal_rotation: Callable = kabsch):
        if weights is not None and len(indices) != len(weights):
            raise RuntimeError("Indices and weights must be of the same length")
        super().__init__(indices)
        self.references = np.asarray(references)
        if weights is None:
            self.weights = np.ones(len(indices)) / len(indices)
        else:
            self.weights = np.asarray(weights)
        self.Q = fitted_positions(self.references, self.weights)
        self.optimal_rotation = optimal_rotation

    @property
    def function(self):
        return lambda r: rmsd(r, self.Q, self.weights, self.optimal_rotation)


def rmsd(r1: np.ndarray, Q: np.ndarray, w_0: np.ndarray, optimal_rotation: Callable):
    """
    Calculate the RMSD respect to a reference using rotation matrix.
    Parameters
    ----------
    r1: np.ndarray
        Atomic positions.
    Q: np.ndarray
        Cartesian coordinates of the reference position of the atoms.
    w_0: np.ndarray
        Weights of the selected atom positions
    """
    P = fitted_positions(r1, w_0)
    U = optimal_rotation(P, Q)
    optimal_P = np.dot(P, U)
    error = np.sqrt(np.sum(np.square(optimal_P - Q)) / P.shape[0])
    return error
```

**Review by pabloferz — CHANGES_REQUESTED** [review]

File: pysages/colvars/orientation.py (hunk 1 above)
Comment (yihengwuKP) [reply pysages/colvars/orientation.py]: cool! I've just commit the changes for this!

**Comment by yihengwuKP** [comment]
not sure why the Docker build test is failing...

**Comment by pabloferz** [comment]
Can you rebase into the latest main?

**Comment by yihengwuKP** [comment]
@pabloferz can we merge this now?

**Comment by pabloferz** [comment]
Yes, sorry about the delay. Thank you for adding these!

**Comment by yihengwuKP** [comment]
no problem, glad to help!

**Pull request merged**